playbooks/tests/tasks/integration.yml

```diff
@@ -6,24 +6,50 @@
     user: name=root generate_ssh_key=yes
 
   - name: generate nova key-pair
-    shell: . /root/stackrc; nova keypair-add turtle-key --pub-key
-           /root/.ssh/id_rsa.pub
+    nova_keypair: name=turtle-key
+                  auth_url={{ endpoints.auth_uri }}
+                  login_tenant_name=admin
+                  login_username=provider_admin
+                  login_password={{ secrets.provider_admin_password }}
+                  public_key=/root/.ssh/id_rsa.pub
+    run_once: true
```

Comment:
This run_once probably isn't necessary, because the play itself only runs on controller[0], so there is only one host.